### PR TITLE
feat: add meta fields and add duplicate processor with eslint-flat-co…

### PR DESCRIPTION
**Pull request by Charlie-Crowdstrike**
…nfig compatible name

https://eslint.org/docs/latest/extend/plugin-migration-flat-config

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-json-files",
-  "version": "4.2.0",
+  "version": "4.2.1",
```
Comment (Charlie-Crowdstrike) [inline package.json:3]: not sure if this should be patch or minor

File: lib/index.js [hunk 2]
```diff
@@ -2,9 +2,20 @@
...
+// required for eslint v9 flat config compatibility
+// https://eslint.org/docs/latest/extend/plugin-migration-flat-config
+module.exports.meta =  {
+  name,
+  version
+},
+
 module.exports.processors = {
-  '.json': require('./processors/json')
+  '.json': require('./processors/json'),
```
Comment (Charlie-Crowdstrike) [inline lib/index.js:17]: leaving the `.json` for backwards compatability

File: package.json (hunk 1 above)
Comment (kellyselden) [reply package.json:3]: This can happen outside the PR.

File: lib/index.js (hunk 2 above)
Comment (kellyselden) [reply lib/index.js:17]: Can you add this as a comment?

File: lib/index.js [hunk 3]
```diff
@@ -2,9 +2,20 @@
...
+  // dot prefix is no longer allowed in eslint v9
+  // https://eslint.org/docs/latest/extend/plugin-migration-flat-config
```
Comment (kellyselden) [inline lib/index.js:19]:
```suggestion
  // https://eslint.org/docs/latest/extend/plugin-migration-flat-config#migrating-processors-for-flat-config
```

File: lib/index.js [hunk 4]
```diff
@@ -2,9 +2,20 @@
 
 const requireIndex = require('requireindex');
 const path = require('path');
+const { name, version } = require('../package.json');
 
 module.exports.rules = requireIndex(path.join(__dirname, 'rules'));
 
+// required for eslint v9 flat config compatibility
```
Comment (kellyselden) [inline lib/index.js:9]: I'm only reading it being optional or recommended. Is it required?

Comment (Charlie-Crowdstrike) [reply lib/index.js:9]: optional for our current use case but required if we ever want to use the 

`-cache and --print-config command line options.`

Comment (kellyselden) [reply lib/index.js:9]: Can you add that to the comment, and link to https://eslint.org/docs/latest/extend/plugin-migration-flat-config#adding-plugin-meta-information also?

**Comment by kellyselden** [comment]
released in eslint-plugin-json-files@4.3.0

**Pull request merged**